framework/CodeInterfaces/RAVEN/RAVENInterface.py

```diff
@@ -186,26 +186,28 @@ def createNewInput(self,currentInputFiles,oriInputFiles,samplerType,**Kwargs):
     # get sampled variables
     modifDict = Kwargs['SampledVars']
     # check if there are noscalar variables
-    noscalarVars, totSizeExpected = {}, 0
+    vectorVars = {}
+    totSizeExpected = 0
     for var, value in modifDict.items():
       if np.asarray(value).size > 1:
-        noscalarVars[var] = np.asarray(value)
-        totSizeExpected += noscalarVars[var].size
-    if len(noscalarVars) > 0 and not self.hasMethods['noscalar']:
-      raise IOError(self.printTag+' ERROR: No scalar variables ('+','.join(noscalarVars.keys())
+        vectorVars[var] = np.asarray(value)
+        totSizeExpected += vectorVars[var].size
+    if len(vectorVars) > 0 and not self.hasMethods['noscalar']:
+      raise IOError(self.printTag+' ERROR: No scalar variables ('+','.join(vectorVars.keys())
                                   + ') have been detected but no convertNotScalarSampledVariables has been inputted!')
     # check if ext module has been inputted
     if self.hasMethods['noscalar'] or self.hasMethods['scalar']:
       extModForVarsManipulation = utils.importFromPath(self.extModForVarsManipulationPath)
     if self.hasMethods['noscalar']:
-      if len(noscalarVars) > 0:
-        toPopOut = noscalarVars.keys()
+      if len(vectorVars) > 0:
+        toPopOut = vectorVars.keys()
         try:
-          newVars = extModForVarsManipulation.convertNotScalarSampledVariables(noscalarVars)
+          newVars = extModForVarsManipulation.convertNotScalarSampledVariables(vectorVars)
           if type(newVars).__name__ != 'dict':
             raise IOError(self.printTag+' ERROR: convertNotScalarSampledVariables must return a dictionary!')
-          if len(newVars) != totSizeExpected:
-            raise IOError(self.printTag+' ERROR: The total number of variables expected from method convertNotScalarSampledVariables is "'+str(totSizeExpected)+'". Got:"'+str(len(newVars))+'"!')
+          # DEBUGG this is failing b/c Index and Variable both being counted!
+          #if len(newVars) != totSizeExpected:
+          #  raise IOError(self.printTag+' ERROR: The total number of variables expected from method convertNotScalarSampledVariables is "'+str(totSizeExpected)+'". Got:"'+str(len(newVars))+'"!')
```

Comment:
This check should not be skipped. In DataObject, if there is a index variable in the <Input> or <Output> we pop them out right? Should not be enough to reactivate this check?

Reply:
They both show up in the modiDict.  I tried to think of a way to work around this, but I didn't come up with anything.

Reply:
so probably I do not understand your explanation in the commented code

Reply:
In the case I tested, I have a vector variable `Demand_time_net` being passed to the inner RAVEN.  It depends on `Time1`.  The interface was calculating an expected number of entries a 50, when I only wanted to send in 25, since I wanted to send in `Demand_time_net`, and `Time1` doesn't get set in the inner RAVEN input file.

Reply:
mmm... I see... I do not like to remove the check...but I approve the modification for now but I would like to find a solution for this...